jetty-security/src/test/java/org/eclipse/jetty/security/ConstraintTest.java

```diff
@@ -1189,6 +1191,107 @@ public void testFormRedirect() throws Exception
+            "Cookie: JSESSIONID=" + sessionId + "\r\n" +
+            "\r\n");
+        assertThat(response, startsWith("HTTP/1.1 200 OK"));
+
+        // check max interval has not been updated
+        assertThat(_sessionhandler.getSession(_sessionhandler.getSessionIdManager().getId(sessionId)).getMaxInactiveInterval(), is(UNAUTH_SECONDS));
+
+        // Now test with changes to the session
+        _server.stop();
+        _sessionhandler.setMaxInactiveInterval(UNAUTH_SECONDS);
+        _security.setSessionMaxInactiveIntervalOnAuthentication(AUTH_SECONDS);
+        _security.setSessionRenewedOnAuthentication(true);
+        _server.start();
+
+        response = _connector.getResponse("GET /ctx/auth/info HTTP/1.0\r\n\r\n");
+        assertThat(response, containsString(" 302 Found"));
+        assertThat(response, containsString("/ctx/testLoginPage"));
+        assertThat(response, containsString("JSESSIONID="));
+        sessionId = response.substring(response.indexOf("JSESSIONID=") + 11, response.indexOf("; Path=/ctx"));
+
+        response = _connector.getResponse("GET /ctx/testLoginPage HTTP/1.0\r\n" +
+            "Cookie: JSESSIONID=" + sessionId + "\r\n" +
+            "\r\n");
+        assertThat(response, containsString(" 200 OK"));
+        assertThat(response, containsString("URI=/ctx/testLoginPage"));
+        assertThat(response, not(containsString("JSESSIONID=" + sessionId)));
+
+        response = _connector.getResponse("POST /ctx/j_security_check HTTP/1.0\r\n" +
+            "Cookie: JSESSIONID=" + sessionId + "\r\n" +
+            "Content-Type: application/x-www-form-urlencoded\r\n" +
+            "Content-Length: 35\r\n" +
+            "\r\n" +
+            "j_username=user&j_password=password");
+        assertThat(response, startsWith("HTTP/1.1 302 "));
+        assertThat(response, containsString("Location"));
+        assertThat(response, containsString("/ctx/auth/info"));
+        assertThat(response, containsString("JSESSIONID="));
+        assertThat(response, not(containsString("JSESSIONID=" + sessionId)));
+        String oldId = sessionId;
+        sessionId = response.substring(response.indexOf("JSESSIONID=") + 11, response.indexOf("; Path=/ctx"));
+
+        // check old session ID no longer works changed.
+        response = _connector.getResponse("GET /ctx/auth/info HTTP/1.0\r\n" +
+            "Cookie: JSESSIONID=" + oldId + "\r\n" +
+            "\r\n");
+        assertThat(response, startsWith("HTTP/1.1 302 Found"));
+
+        // check new session ID is authenticated.
+        response = _connector.getResponse("GET /ctx/auth/info HTTP/1.0\r\n" +
+            "Cookie: JSESSIONID=" + sessionId + "\r\n" +
+            "\r\n");
+        assertThat(response, startsWith("HTTP/1.1 200 OK"));
+
+        // Check max interval has been updated
+        assertThat(_sessionhandler.getSession(_sessionhandler.getSessionIdManager().getId(sessionId)).getMaxInactiveInterval(), is(AUTH_SECONDS));
+    }
```

Comment:
Strictly speaking, there should be another couple of test cases where we renew the sessionId but don't renew the timeout, or we don't renew the sessionId but we do renew the timeout.

Reply:
done